### PR TITLE
fix(notify): restrict `schedule_wrap` to `refresh()`

### PR DESCRIPTION
**Pull request by przepompownia**
It allows adding notification and returning its id in fast event.

- [x] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [x] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lua/mini/notify.lua [hunk 1]
```diff
@@ -309,10 +308,8 @@ MiniNotify.make_notify = function(opts)
 
     local id = MiniNotify.add(msg, level_name, level_data.hl_group)
     vim.defer_fn(function() MiniNotify.remove(id) end, level_data.duration)
-  end
-  return function(msg, level)
-    if not vim.in_fast_event() then return notify(msg, level) end
-    vim.schedule(function() notify(msg, level) end)
+
+    return id
```
Comment (przepompownia) [inline lua/mini/notify.lua:312]: If compliance with `vim.notify` is important I can remove it - it's not essential and the user can write custom replacement of `make_notify`.

**Comment by echasnovski** [comment]
Thanks for the PR!

This indeed caused problems once or twice before (but allowed to find issues with other modules, but not really important).

I think I'd prefer to have `refresh()` itself schedule its effects depending on whether currently inside fast event. Not sure about the best approach yet, though.

**Comment by przepompownia** [comment]
> Thanks for the PR!
> 
> This indeed caused problems once or twice before (but allowed to find issues with other modules, but not really important).

Probably it doesn't matter much assuming using `make_notify` wrapper. I found the problem when I tried to call `update()` inside `vim.ui_attach` callback (probably especially if `return_prompt` event causes waiting for `<CR>` input). Example (with modified mini.notify): https://github.com/przepompownia/mini-ui-messages-example

> I think I'd prefer to have `refresh()` itself schedule its effects depending on whether currently inside fast event. Not sure about the best approach yet, though.

Do you mean the `if not vim.in_fast_event()` logic could be placed inside `M.refresh()`? I considered it at the start but didn't found any simpler approach than the current.

**Comment by echasnovski** [comment]
> > I think I'd prefer to have `refresh()` itself schedule its effects depending on whether currently inside fast event. Not sure about the best approach yet, though.
> 
> Do you mean the `if not vim.in_fast_event()` logic could be placed inside `M.refresh()`? I considered it at the start but didn't found any simpler approach than the current.

Yeah, so that all current direct `refresh()` usages stay as is. I'll take a look later.

**Comment by echasnovski** [comment]
Calling `MiniNotify.refresh()` directly inside fast events now should be possible on latest `main`.

**Pull request closed without merging**